### PR TITLE
Maps numbers to names in TSGroupModel

### PR DESCRIPTION
**Pull request by mdouglass**
# Use an injected ContactsManager to look up proper user names for phone numbers in [TSGroupModel getInfoStringAboutUpdateTo].

Fixes WhisperSystems/Signal-iOS#757

// FREEBIE

### CODE REVIEW

**Comment by mdouglass** [comment]
This is my first commit to this project, so feel free to beat up on it.  I think I've properly following the contribution guidelines and I've submitted a CLA. Let me know if I missing anything else.

File: src/Messages/TSGroupModel.h [hunk 1]
```diff
@@ -19,7 +20,7 @@
 
 - (BOOL)isEqual:(id)other;
 - (BOOL)isEqualToGroupModel:(TSGroupModel *)model;
-- (NSString *)getInfoStringAboutUpdateTo:(TSGroupModel *)model;
+- (NSString *)getInfoStringAboutUpdateTo:(TSGroupModel *)model withContactsManager:(id<ContactsManagerProtocol>)contactsManager;
```
Comment (michaelkirk) [inline src/Messages/TSGroupModel.h]: style nit, name the parameter without the preposition, just `contactsManager:(id<ContactsManagerProtocol>)contactsManager`

**Review by michaelkirk — COMMENTED** [review]
Thanks for this.

This works when someone else updates the group.

Beyond your improvement here, it would be good to address what happens when _I'm_ the one updating the group. As it is, I see just "Group Updated" in that case.

See: 
`- (void)updateGroupModelTo:(TSGroupModel *)newGroupModel` in MessagesViewController.m for a similar method.

Also, if you're into using cool functions like `map` and injecting dependencies, there's plenty more work to be done in Signal iOS. =)

**Comment by mdouglass** [comment]
I'll update the parameter name.  
Should I make the other change in this PR or open another one for that issue?
I'm definitely planning to do more work on Signal-iOS.  Wanted to pick something small to get a foot in the door :)

**Comment by michaelkirk** [comment]
Getting group updates made locally to show these details will likely require changes to Signal-iOS, so it will necessitate a PR over there.

I was thinking it made sense to wait on merging this one so you could make sure any shared code works in both cases.

**Comment by mdouglass** [comment]
That makes sense, I'll take a look.

**Comment by michaelkirk** [comment]
Here's where we're currently handling it
https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/view%20controllers/MessagesViewController.m#L2011

Seemingly as a result of:
https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/view%20controllers/MessagesViewController.m#L2041

**Comment by mdouglass** [comment]
Ack, something went wrong with that merge before I committed it. I'll have to redo it.

**Comment by michaelkirk** [comment]
Feel free to force push your own branches in the future.

**Comment by mdouglass** [comment]
Good to know.  Never quite sure when that's ok with git.  Too much time spent in p4.

**Pull request closed without merging**